### DISCUSSION TITLE
Post Requets not working CSRF Failed: CSRF token missing.

### DISCUSSION THREAD

**Discussion by philipphomberger** [encode/django-rest-framework#9246] [Potential Issue] [answered]
I try using Django Restframework together with VueJS and axion. 
But always I get the MSG: 
CSRF Failed: CSRF token missing.

But my Header in the frontend looks correct. And in the developer tools the cockie is correct loading into the header.
{
  "Accept": "application/json, text/plain, */*",
  "Content-Type": "application/json",
  "X-CSRFToken": "*******"
}

my csrf settings in django settings.py

CSRF_COOKIE_NAME = "csrftoken"
CSRF_HEADER_NAME = 'X-CSRFTOKEN'
CSRF_TRUSTED_ORIGIN = ['http://*.127.0.0.1', 'http://*.localhost']
CSRF_COOKIE_HTTPONLY = False 
CSRF_COOKIE_SECURE= False

I have no problems with the get requests. Only than it come to POST, PUT, DELETE.

Thank you for your advice.

With regards 
Philipp Homberger

**Comment by FaridMalekpour** [comment]
is the CSRF middleware included in Django project's `MIDDLEWARE` setting in `settings.py`?

```python
MIDDLEWARE = [
    ...
    'django.middleware.csrf.CsrfViewMiddleware',
    ...
]

**Reply by philipphomberger** [reply, marked as answer]
Hi, there was a cache problem in my Browser or so. So the HTTP Only Attribute was still set to true. Because of this my Java Script client could not use the Csrf toke.

**Reply by FaridMalekpour** [reply]
Glad to hear the problem is solved🤝

**Reply by MatinMakiabadi** [reply]
> Hi, there was a cache problem in my Browser or so. So the HTTP Only Attribute was still set to true. Because of this my Java Script client could not use the Csrf toke.

I am so grateful that you post this solution, I was having such a fruastrating time.

**Reply by Beki78** [reply]
tnx mate I think I was having the same problem when I changed the browser it worked like a charm😌

**Comment by MatinMakiabadi** [comment]
I was banging my head on the table for 2 hours! 😹
